### ISSUE TITLE
gdax not found for `fetchOrder` on Php

### ISSUE THREAD

**Issue by koushik72** [ccxt/ccxt#2906]
There is an issue that I have found for gdax on php version.
`fetchOrder` for `gdax` doesn't work and says that `gdax` not found.
I tried to check if gdax has `fetchOrder` and found out that it was true.
It's working fine for 4 other exchanges that I am working with currently.
Please fix it.
Thank you.

**Comment by kroitor** [comment]
Can you post your code to reproduce the issue? (please make it short, but copy the entire program, we need to see the instantiation code and the erroneous call).

**Comment by koushik72** [comment]
Hi, thank you for the reply.
`foreach ($unProcessedOrders as $uo) {
            $exchangeName = strtolower($uo->exchange_name);
            $ccxtClassName = '\\ccxt\\' . $exchangeName;
            $exchange = new $ccxtClassName();
            $exchange->apiKey = $uo->api_key;
            $exchange->secret = $uo->api_secret;
            if ($exchangeName == 'gdax') {
                $exchange->password = $uo->api_password;
            }
            $extraParams = [];
            if ($exchangeName == 'binance') {
                $extraParams = [
                    'timestamp' => $this->getCurrentMicroSecs(),
                    'recvWindow' => 20000
                ];
            }
            try {
                $order = $exchange->fetchOrder($uo->exchange_order_id, $uo->ccxt_symbol, $extraParams);
                $order['exchange_id'] = $uo->exchange_id;
                $order['order_id'] = $uo->order_id;
                $orders[] = $order;
            } catch (\Exception $e) {
                $error[] = [
                    'exchange_name' => $uo->exchange_name,
                    'order_id' => $uo->order_id,
                    'error' => $e->getMessage()
                ];
                return $e->getMessage();
            }
        }`
In the above code, I am basically working with 5 exchanges and all are working fine except for Gdax.
My language is `Php` and version is `7.2.5`. I am using it within `Laravel 5.6.15`.  ccxt version is `^1.11`. 
The method is `fetchOrder` and exchange is `gdax`.
Thank you.

**Comment by kroitor** [comment]
> ccxt version is ^1.11.

Can you please tell the version exactly? 1.11 is several months old (too old), the current active branch is 1.14+, you should really upgrade it first, then retry the code again with the most recent version. Can you verify that?

**Comment by koushik72** [comment]
Hi, thank you for the reply.
I did update to the latest version and it's still the same and says that `gdax not found`.
Please fix it.
Thank you.

**Comment by kroitor** [comment]
Fix what? ) How do you link against it? Basically, if it's working for 99% of users, then there's a strong indication that this is a problem with the setup. Probably, some classes overriding gdax or a namespace conflict. Can you post the smallest snippet of your code to reproduce it?

```PHP
<?php
include 'ccxt.php';
$exchange = new \ccxt\gdax ();
?>
```

Alternatively, try another exchange to see if it works:

```PHP
include 'ccxt.php';
$exchange = new \ccxt\binance ();
```

↑ Does any of these simplest examples work for you?

**Comment by kroitor** [comment]
Ok, nevermind my previous post. Will look into and will get back to you on this shortly.

**Comment by kroitor** [comment]
Can you try the same code with an underscore notation for `fetch_order`?

```PHP
foreach ($unProcessedOrders as $uo) {

            $exchangeName = strtolower($uo->exchange_name);
            $ccxtClassName = '\\ccxt\\' . $exchangeName;
            $exchange = new $ccxtClassName();
            $exchange->apiKey = $uo->api_key;
            $exchange->secret = $uo->api_secret;
            if ($exchangeName == 'gdax') {
                $exchange->password = $uo->api_password;
            }
            $extraParams = [];
            if ($exchangeName == 'binance') {
                $extraParams = [
                    'timestamp' => $this->getCurrentMicroSecs(),
                    'recvWindow' => 20000
                ];
            }
            try {
                $order = $exchange->fetch_order($uo->exchange_order_id, $uo->ccxt_symbol, $extraParams);
                $order['exchange_id'] = $uo->exchange_id;
                $order['order_id'] = $uo->order_id;
                $orders[] = $order;
            } catch (\Exception $e) {
                $error[] = [
                    'exchange_name' => $uo->exchange_name,
                    'order_id' => $uo->order_id,
                    'error' => $e->getMessage()
                ];
                return $e->getMessage();
            }
        }
```

↑  Does it work for you this way?

**Comment by koushik72** [comment]
Hi, thank you for your reply and sorry for my late reply.
I tried with the underscore notation and it's the same.

**Comment by koushik72** [comment]
It's solved now.
I tried with another order id and it worked. 
I tried with three order ids and for two it worked as expected and for one, it didn't work. 
When I paid more attention to my laravel error page, I found out that it said `throw new OrderNotFound ($error);`.
So, I think it didn't find that order.
I think the error msg should be more specific so that it doesn't confuse.
Thank you for this amazing library btw.
